pkg/metrics/metrics.go

```diff
@@ -224,3 +257,26 @@ func updateServiceMonitors(cfg *rest.Config, namespace string) error {
 	}
 	return nil
 }
+
+// removeStaleResources deletes any stale resources that could be left behind during the
+// upgrade process due to the renaming of resources between current and previous operator version
+func (mc *metricsConfig) removeStaleResources(ctx context.Context, cfg *rest.Config, namespace string) error {
```

Comment:
You need to call out that you are deleting the old resource in the commit message.